src/Units/UnitContext.java

```diff
@@ -171,7 +171,10 @@ public void setWeapon(WeaponModel w)
 
   public int getCostTotal()
   {
-    return (int) (costBase*costMultiplier)+costShift;
+    int scaledCost = (costBase + costShift) * costRatio / UnitModel.DEFAULT_STAT_RATIO;
```

Comment:
Should this (and the corresponding places in Bear/Bull) be `(costBase + costShift) * (costRatio / UnitModel.DEFAULT_STAT_RATIO);`

Reply:
I'm not aware of any corresponding places in Bear/Bull. Do you mean the income change?
Adding those parentheses is a terrible idea because this is all integer math:
`(costBase + costShift) * (costRatio / UnitModel.DEFAULT_STAT_RATIO)`
`(1000 + 1000) * (50 / 100)`
`(2000) * (0)`
`0`
vs
`(costBase + costShift) * costRatio / UnitModel.DEFAULT_STAT_RATIO`
`(1000 + 1000) * 50 / 100`
`(2000) * 50 / 100`
`100000 / 100`
`1000`

Reply:
So it is; fair enough.